pyro/contrib/tracking/hashing.py

```diff
@@ -0,0 +1,130 @@
+from __future__ import absolute_import, division, print_function
+
+import itertools
+from collections import defaultdict
+from numbers import Number
+
+
+class LSH(object):
+    """
+    Implements locality-sensitive hashing.
+
+
+    Allows to efficiently find neighbours of a point. Provides 2 guarantees:
+
+    -   Difference between coordinates of points not returned by :meth:`nearby`
+        and input point is larger than ``radius``.
+    -   Difference between coordinates of points returned by :meth:`nearby` and
+        input point is smaller than 2 ``radius``.
+
+    Example:
+
+        >>> radius = 1
+        >>> lsh = LSH(radius)
+        >>> a = torch.tensor([-0.51, -0.51]) # hash(a)=(-1,-1)
+        >>> b = torch.tensor([-0.49, -0.49]) # hash(a)=(0,0)
+        >>> c = torch.tensor([1.0, 1.0]) # hash(b)=(1,1)
+        >>> lsh.add('a', a)
+        >>> lsh.add('b', b)
+        >>> lsh.add('c', c)
+        >>> print(lsh.nearby('a')) # {'b'}, even though c is within 2radius of a
+        >>> print(lsh.nearby('b')) # {'a','c'}
+        >>> lsh.remove('b')
+        >>> print(lsh.nearby('a')) # {}
```

Comment:
@alicanb - This is failing the doctest. You can run `make docs` to replicate the failure. You can just remove the prints and append the output below for the tests to pass. Let me know if you need any help fixing this. e.g.

```
>>> lsh.nearby('a')
set(['b'])
```

Reply:
Thanks @neerajprad , weird- I can build successfully on my machine... Anyways I pushed a fix, I'll ask you if this doesn't work.